### PR TITLE
Fix shapes in comments of nmt_with_attention.ipynb

### PR DESCRIPTION
**Pull request by RayzhiYang**
It is a bit misleading and confusing that the output shape of decoder is currently commented as `(batch_size * max_length, vocab)`. However the correct shape should be `(batch_size * 1, vocab)`, since the input x of GRU layer has shape == `(batch_size, 1, embedding_dim + hidden_size)`.

### CODE REVIEW

**Comment by SayHiRay** [comment]
@googlebot I signed it

**Comment by SayHiRay** [comment]
It also seems confusing to me that in the `evaluate` method, `tf.multinomial(tf.exp(predictions), num_samples=1)` is used to sample a prediction from the `predictions` logits. The usage of `tf.exp` seems redundant here, since `predictions` itself is the logits output of a dense layer in decoder.

**Comment by case540** [comment]
Anyone of the reviewers I added know who wrote or knows about the RNN attention models?

**Comment by lamberta** [comment]
@yashk2810

**Review by MarkDaoust — DISMISSED** [review]

**Comment by yashk2810** [comment]
@SayHiRay Nice catch. Would you mind making the same change in the text generation notebook? Its in the generative examples folder. And also other notebooks?

**Comment by MarkDaoust** [comment]
`tf.multinomial` does expect un-normalized log-probabilities.
The `Decode` model is only ever called with input batch_size of 1.

So both of these changes look correct.

**Comment by SayHiRay** [comment]
@yashk2810 @MarkDaoust Thanks for reviewing and mentioning about other notebooks. I have checked the usage of `tf.multinomial` in text_generation.ipynb and image_captioning_with_attention.ipynb, and found same mistakes. I have committed the changes. Please help take a look again. Thanks for help!

**Comment by yashk2810** [comment]
LGTM. Thanks for the fix 😀

**Review by MarkDaoust — APPROVED** [review]

**Comment by yashk2810** [comment]
@MarkDaoust I think this needs a "ready to pull" label

**Comment by MarkDaoust** [comment]
Yes!

**Comment by yashk2810** [comment]
Does this get merged automatically? I think it should get merged by the copybara bot right?

**Comment by case540** [comment]
Yeah, its currently in the process of being merged. Some internal test flakes have slowed it down

**Comment by yashk2810** [comment]
I think the internal tests are failing because an owner hasn't approved it. It happened with me before.

**Comment by case540** [comment]
Just looks like a test flake to me. Tests are re-running now

**Comment by yashk2810** [comment]
Sounds good 😀

**Comment by SayHiRay** [comment]
Thanks a lot for helping with merging! 👏
